File: be/src/main/java/backend/service/MeetingService.java
package backend.service;

import backend.entity.Meeting;
import java.util.*;

public interface MeetingService {

    Meeting getById(long id);
    void addMeeting(Meeting meeting);
    void delete(long id);
    void deleteForCur(Meeting meeting, String login);
    List<Meeting> getAll();
    List<Meeting> getByDate(Calendar date);
}

File: frontend/src/app/service/meeting.service.ts
import { Injectable } from '@angular/core';
import {Observable} from 'rxjs';
import {Meeting} from '../../model/Meeting';
import {HttpClient, HttpHeaders} from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class MeetingService {

  constructor(private http: HttpClient) { }
  public addMeeting(meeting: Meeting): Observable<any> {
    return this.http.post('api1/meeting/', meeting);
  }
  getByDate(date): Observable<any> {
    return this.http.post('api1/meeting/date', date);
  }
  public delete(id: string) {
    return this.http.delete('api1/meeting/' + id);
  }
  public deleteForCur(meeting: Meeting, login: string) {
    return this.http.put('api1/meeting/' + login, meeting);
  }
}
